cr.h

```diff
@@ -1625,8 +1626,18 @@ static bool cr_plugin_load_internal(cr_plugin &ctx, bool rollback) {
         }
 
         auto new_version = ctx.version + (rollback ? 0 : 1);
-        const auto new_file = cr_version_path(file, new_version, p->temppath);
-        if (!rollback) {
+        auto new_file = cr_version_path(file, new_version, p->temppath);
+        if (rollback) {
+            // Don't rollback to this version again, if it crashes.
+            ctx.last_version = ctx.version > 0 ? ctx.version - 1 : 0;
+        } else {
+            // Save current version for rollback.
+            ctx.last_version = ctx.version;
+            while(cr_exists(new_file)) {
```

Comment:
Overall looks good, but I have a bit of concern about this while here. We could probably save the latest achieved version too, so we can skip directly to it. And then change `last_version` to `last_working_version`. What do you think?
Nitpick: put a space in `while (`

Reply:
How about `next_version` and `last_working_version`?  Also should they be in the private `cr_internal` or stay in `cr_plugin`?

Reply:
Also do you think we should keep the `while (..` and skip old plugin version from previous host runs?  If we track the next version, then we don't really need to check for existing plugin files.

Reply:
`next_version` and `last_working_version` should be good, and yes we should not need the while anymore! Thanks!